content/2022-04-13-our-secure-supply-chain-obsession.md

```diff
@@ -0,0 +1,383 @@
+always exist that a GitHub account has been compromised. Verified commits are
+cryptographically signed with a GPG key, thereby offering stronger assurances of
+the author's identity. Is it foolproof? Of course not. Keys can be stolen. But
+since we've already established that eternal vigilance is the cost to secure
+one's software supply chain, every incremental improvement can be counted as a
+small win.
+
+To be fully transparent, there's an undeniable downside to our policy of
+insisting on verified commits -- and that is many would-be contributors are
+presently unfamiliar with the steps required to cryptographically sign their
+commits and we believe that has stymied our efforts to grow our contributor
+base. While this is unfortunate, we believe that this effect is a temporary one.
+Since we expect the industry-wide obsession over software supply chain security
+will continue to grow, we also expect most open source projects will eventually
+insist on verified commits, and as that practice trends toward ubiquity, it will
+become more common for would-be contributors to be familiar with the required
+steps. In the near term, we've attempted to mitigate the adverse effects of this
+policy by producing an instructional video on the topic that we share with any
+would-be contributors who open pull requests containing unverified commits.
+
+## Monitoring Our Dependencies
+
+Another step we took some time ago in an effort to secure our software supply
+chain was to enable [Dependabot](https://github.com/dependabot) on all of our
+source code repositories. This is quick and easy to do, and besides alerting
+maintainers to vulnerabilities in third-party dependencies, it also
+automatically opens PRs to upgrade those same dependencies once a patch becomes
+available. If you're not already doing this with your own repositories, we
+suggest you start immediately.
+
+But our effort to monitor our dependencies didn't end with Dependabot; it merely
+began there. For TypeScript-based components of Brigade and related projects,
+we've also integrated a `yarn audit` into our CI processes. While this mostly
+surfaces the same vulnerabilities as Dependabot, integrating it directly into
+our CI processes means we have the opportunity to detect new vulnerable
+dependencies _before_ they're ever committed to our main branch, but with that
+being said, the maintainers do not strictly require `yarn audit` to pass in
+order for a PR to be merged, because in many cases, the PR we're reviewing has
+_not_ introduced new third-party dependencies and isn't remotely responsible for
+the failed audit -- and this underscores a critically important point: Software
+that passes every single scan or audit you can throw at it _today_ may fail
+_tomorrow_, without a single change having been made in the interim. This can
+happen simply because new vulnerabilities have been _reported_ in the interim.
+_Stale software is vulnerable software,_ so I will repeat -- the cost of
```

Comment:
Can we do "quotes" in the blog post? I like this call out and think it's worth highlighting.

Reply:
Like giant block quotes? I'm not sure if the theme we're using supports that or how to do it, but I can find out.

I agree this is a very important point that deserves to be highlighted.

Reply:
Think I got it.

Reply:
https://deploy-preview-46--blog-brigade-sh.netlify.app/2022-04-14-our-secure-supply-chain-obsession/